src/go/rpk/pkg/config/params.go

```diff
@@ -552,3 +568,133 @@ func (c *Config) addUnsetDefaults() {
 		r.AdminAPI.Addresses = []string{"127.0.0.1:9644"}
 	}
 }
+
+// Set allow to set a single configuration property by passing a key value pair
+//
+//   Key:    string containing the yaml property tag, e.g: 'rpk.admin_api'.
+//   Value:  string representation of the value, either single value or partial
+//           representation if a format is passed (yaml / json).
+//   Format: either single, json, or yaml (default: single).
+func (c *Config) Set(key, value, format string) error {
+	if key == "" {
+		return fmt.Errorf("key field must not be empty")
+	}
+	props := strings.Split(key, ".")
+	rv := reflect.ValueOf(c).Elem()
+	found, other, err := getField(props, rv)
+	if err != nil {
+		return err
+	}
+	isOther := other != reflect.Value{}
+
+	field := found
+	if isOther {
+		field = other
+	}
+
+	if field.CanAddr() {
+		i := field.Addr().Interface()
+		in := value
+		switch strings.ToLower(format) {
+		case "yaml", "single", "":
+			if isOther {
+				p := props[len(props)-1]
+				in = fmt.Sprintf("%s: %s", p, value)
+			}
+			err = yaml.Unmarshal([]byte(in), i)
+			if err != nil {
+				return err
+			}
+			return nil
+		case "json":
+			if isOther {
+				p := props[len(props)-1]
+				in = fmt.Sprintf("{%q: %q}", p, value)
+			}
+			err = json.Unmarshal([]byte(in), i)
+			if err != nil {
+				return err
+			}
+			return nil
+		default:
+			return fmt.Errorf("unsupported format %s", format)
+		}
+	}
+	return errors.New("rpk bug, please describe how you encountered this at https://github.com/redpanda-data/redpanda/issues/new?assignees=&labels=kind%2Fbug&template=01_bug_report.md")
+}
+
+// getField deeply search in p for the value that reflect property props.
+func getField(props []string, p reflect.Value) (reflect.Value, reflect.Value, error) {
+	if len(props) == 0 {
+		return p, reflect.Value{}, nil
+	}
+	if p.Kind() == reflect.Slice && p.Len() > 0 {
+		p = p.Index(0)
+	}
+	if p.Kind() == reflect.Slice && p.Len() == 0 {
+		p.Set(reflect.New(p.Type().Elem()))
+	}
+	// If is a nil pointer we assign the zero value, and we reassign p to the
+	// value that p points to
+	if p.Kind() == reflect.Ptr {
+		if p.IsNil() {
+			p.Set(reflect.New(p.Type().Elem()))
+		}
+		p = reflect.Indirect(p)
+	}
+	if p.Kind() == reflect.Struct {
+		newP, other, err := getFieldByTag(props[0], p)
+		if err != nil {
+			return reflect.Value{}, reflect.Value{}, err
+		}
+		// if is "Other" map field, we stop the recursion and return
+		if (other != reflect.Value{}) {
+			// user may try to set deep unmanaged property:
+			// rpk.unmanaged.name = "name"
+			if len(props) > 1 {
+				return reflect.Value{}, reflect.Value{}, fmt.Errorf("cannot set property %v", strings.Join(props, "."))
```

Comment:
`unable to set field %q`?

Reply:
Also, doesn't this error (and the error below) not have the full props list, since we are recursing down it? I think we want the caller of this function to say "unable to set %q", and the error here to just be a bit of extra context.

Reply:
Changing for: unable to set property %q using rpk. 

This error is for someone that is trying to set an rpk unmanaged property:

redpanda.unmanaged.name = rogger

So the complete error should look like this:

Error: unable to set 'redpanda.unmanaged.name': unable to set property 'unmanaged.name' using rpk.

wdyt?

Reply:
first lgtm